ProboCI/probo-image-builder
# Feature/lliss/overhaul

I have completely declared bankruptcy on our puppet setup due to an inability to get the puppet module dependencies to resolve and give me the versions I want. Frankly, I never liked them anyway. This remove all the puppet code and replaces it with Ansible provisioning. This has many advantages

- no dependencies
- reliable ordering of running tasks
- fewer package dependencies within the VM and docker images
- it works

My goal was to reproduce the setup needed to run a probo docker image. These are finicky. They must have a bunch of services installed but not running. I also didn't want to dramatically alter any of the settings that were already in use, since we will eventually switch this out for the main image.

Currently this only created a php5.6 image on ubuntu 14.04. Later we can rework it to create php7 and 7.1.

## To Test
- Ensure `ansible` is installed on your workstation.
- https://docs.ansible.com/ansible/latest/intro_installation.html#latest-releases-on-mac-osx
- Clone the repo and run `vagrant up`
- This should do EVERYTHING
- At the end of the provisioning, a tar file called `ubuntu-14.04-lamp-nightly.tar` should be in the workspace folder.
- Take this tar file and import in into your probo vm.
- Once on the probo vm run `docker load < ubuntu-14.04-lamp-nightly.tar`
- This will load the docker image and allow it to be used.
- Confirm this by running `docker images` and check to see if the image is there.
- Now alter your probo container manager yaml file.
- Change the default image to `defaultImage: 'proboci/ubuntu-14.04-lamp:nightly'`
- Add a new config for the image or alter the existing default one.
- It should look like the example below.
- Restart the container manager `sudo service probo-container-manager restart`
- Run a build that uses the default image or specify the image in your probo.yaml file.
- Ensure the build works like normal.
- On the command line run `docker ps` and ensure that the newly running container is in fact using the new image.
- Celebrate!

```yaml
image: 'proboci/ubuntu-14.04-lamp:nightly'
 # The images used by the host manager
 images:
   'proboci/ubuntu-14.04-lamp:nightly':
     services:
       cleanapache:
         command: 'rm /var/run/apache2/apache2.pid'
       apache:
         command: '/usr/sbin/apache2ctl -D FOREGROUND'
         port: 80
       mysql:
         command: 'mysqld_safe'
       redis:
         command: 'redis-server'
       php_log:
         command: 'tail -F /var/log/php/error.log'
```
